dcc-submission-ui/dictionary/app/scripts/helpers/dictionary-utils.js

```diff
@@ -295,21 +311,22 @@ var dictionaryApp = dictionaryApp || {};
           }
           
           // Now reverse logic to check for completely new file types
-          if (dictTo) {
+          if (dictTo.files) {
             dictTo.files.forEach(function (fileTo) {
               var fileFrom = _self.getFileTypeFromDict(dictFrom, fileTo.name);
               // fileTo is new
               if (!fileFrom) {
                 fileTo.fields.forEach(function (fieldTo) {
                   report.fieldsAdded.push({
                     fileType: fileTo.name,
-                    fieldName: fieldTo.name
+                    fieldName: fieldTo.name,
+                    fieldDiff: jsondiffpatch.formatters.html.format(jsondiffpatch.diff({}, fieldTo))
```

Comment:
Seems to be repeated, so good candidate for encapsulation. Probably `jsondiffpatch.formatters.html.format` should be a directive which uses the service as it produces html.